alibi_detect/cd/cvm.py

```diff
@@ -71,6 +71,8 @@ def __init__(
             input_shape=input_shape,
             data_type=data_type
         )
+        # Set config
+        self._set_config(locals())
```

Comment:
Noticed that in some cases you do `self._set_config(locals())` whilst in others it's
```
inputs = locals().copy()
self._set_config(inputs)
```

What's the difference?

Reply:
In the non-backend specific detectors we simply have `self._set_config(locals())` right after `__init__`, to ensure we get the "fresh" unedited args/kwargs. 

In the backend-specific detectors we can't call `self._set_config()` until `self.meta` has been set, therefore we have to call it at the end of the `__init__` method. `inputs = locals()` is still called immediately after `__init__` so that we capture the original args/kwargs (e.g. before `backend` etc edited). The `.copy()` was added as `items` being mutable meant it's items were modified before `_set_config` is reached.

The above works, but unsure if its the best approach... Would need documenting in `CONTRIBUTING.md` at the very least...

Reply:
Yea, needs documenting.